featurebyte/service/query_cache_manager.py

```diff
@@ -70,14 +71,34 @@ async def get_or_cache_table(
         """
         query = sql_to_string(table_expr, source_type=session.source_type)
 
+        def _get_table_details(_table_name: str) -> TableDetails:
+            return TableDetails(
+                database_name=session.database_name,
+                schema_name=session.schema_name,
+                table_name=_table_name,
+            )
+
         cached_table_name = await self.get_cached_table(feature_store_id, query)
+
+        # Check validity of the cached table
+        if cached_table_name is not None:
+            try:
+                await session.table_exists(_get_table_details(cached_table_name))
+            except session._no_schema_error:
```

Comment:
I think `table_exists` already catches the exception and returns `False` if table does not exist?

Reply:
Indeed, will fix this.